notification-rule-config-service-impl/src/main/java/org/hypertrace/notification/config/service/NotificationRuleConfigServiceRequestValidator.java

```diff
@@ -41,23 +43,9 @@ public void validateUpdateNotificationRuleRequest(
   }
 
   private void validateNonDuplicateNotificationRuleOrThrow(
-      String ruleName, List<NotificationRule> existingNotificationRules) {
+      @Nullable String id, String ruleName, List<NotificationRule> existingNotificationRules) {
     for (NotificationRule existingNotificationRule : existingNotificationRules) {
-      if (existingNotificationRule
-          .getNotificationRuleMutableData()
-          .getRuleName()
-          .equals(ruleName)) {
-        throw Status.ALREADY_EXISTS
-            .withDescription("Notification Rule with the same name already exists.")
-            .asRuntimeException();
-      }
-    }
-  }
-
-  private void validateNonDuplicateNotificationRuleOrThrow(
-      String id, String ruleName, List<NotificationRule> existingNotificationRules) {
-    for (NotificationRule existingNotificationRule : existingNotificationRules) {
-      if (!existingNotificationRule.getId().equals(id)
+      if ((Objects.isNull(id) || !existingNotificationRule.getId().equals(id))
```

Comment:
You could put an explicit null check here if you want, but it's not really needed since no existing notification would have a null ID by definition and comparison an existing ID to null is valid. Basically, it doesn't change the logic and makes it a little more confusing to read.